ci/openstack/provision.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+set -euo pipefail
+
+source ci/openstack/vars.sh
+if [ "${RUN_OPENSTACK_CI:-}" != "true" ]; then
+    echo RUN_OPENSTACK_CI is set to false, skipping the openstack end to end test.
+    exit
+fi
+
+if [ "${CI_OVER_CAPACITY:-}" == "true" ]; then
+    echo the CI is over capacity, skipping the end-end test.
+    exit 1
+fi
+
+echo "Keypair name: $KEYPAIR_NAME"
+
+openstack keypair create "$KEYPAIR_NAME" > ~/.ssh/id_rsa
+chmod 600 ~/.ssh/id_rsa
+
+echo CONFIGURE THE INVENTORY
+
+export INVENTORY="$PWD/playbooks/provisioning/openstack/sample-inventory"
+
+mv "$INVENTORY"/ansible.cfg .
+
+sed -i "s/^openstack_ssh_public_key.*/openstack_ssh_public_key: $KEYPAIR_NAME/" "$INVENTORY"/group_vars/all.yml
+sed -i 's/^openstack_external_network_name.*/openstack_external_network_name: "38.145.32.0\/22"/' "$INVENTORY"/group_vars/all.yml
+sed -i 's/^openstack_default_image_name.*/openstack_default_image_name: "CentOS-7-x86_64-GenericCloud-1703"/' "$INVENTORY"/group_vars/all.yml
+sed -i 's/^openstack_num_nodes.*/openstack_num_nodes: 1/' "$INVENTORY"/group_vars/all.yml
+sed -i "s/^env_id.*/env_id: $ENV_ID/" "$INVENTORY"/group_vars/all.yml
+
+PUBLIC_IP="$(curl --silent https://api.ipify.org)"
+echo "node_ingress_cidr: $PUBLIC_IP/32" >> "$INVENTORY"/group_vars/all.yml
+echo "manage_packages: False" >> "$INVENTORY"/group_vars/all.yml
+echo "ephemeral_volumes: True" >> "$INVENTORY"/group_vars/all.yml
+
+# TODO(shadower): this just started to break. If this fix works, let's remove
+# it from the sample inventory proper.
+sed -i 's/openshift_release/#openshift_release/' "$INVENTORY"/group_vars/OSEv3.yml
```

Comment:
note, these can be passed the command line -e args as well. Simplifies the code, removes seds :)